src/app/usecases/TagUseCase.js

```diff
@@ -1,34 +1,46 @@
 import {useState} from "react";
+import {getStringHash} from "../../utils/utils";
 
 
-const useTags = () => {
-    const [tags, setTags] = useState([]);
+const useTextInputContent = (initialTextInputContent = []) => {
+    const [textInputContents, setTextInputContents] = useState(initialTextInputContent);
 
-    const addTag = tag => {
-        if (tagAlreadyExists(tag, tags) || isIllegalText(tag)) {
-            console.log(`[addTag] tag ${tag} not added.`);
+    const addTextInputContent = content => {
+        if (contentAlreadyExists(content, textInputContents) || isIllegalText(content)) {
+            console.log(`[addTextInputContent] content ${content} not added.`);
             return;
```

Comment:
NIT: What is this for ```return;```

Reply:
The code is written by @nyngwang 😢 Let me ask him.

Reply:
@Wally5077 This function is `add...`. The condition means "If the content already exists" or "if the content does not exist, but it contains illegal content". We should return so line 17 won't be executed.

Reply:
@nyngwang It's weird.
You should write an ```if``` condition for line 17 instead of using ```return```.